### PR TITLE
Add Span.updateName().

### PR DESCRIPTION
**Pull request by carlosalberto**

### CODE REVIEW

**Comment by carlosalberto** [comment]
Addresses #20

**Comment by carlosalberto** [comment]
Addresses #23

File: api/src/main/java/openconsensus/trace/Span.java [hunk 1]
```diff
@@ -117,6 +117,16 @@
    */
   public abstract void setStatus(Status status);
 
+  /**
+   * Updates the {@code Span} name.
+   *
+   * <p>If used, this will override the name provided via {@code SpanBuilder}.
```
Comment (SergeyKanzhelev) [inline api/src/main/java/openconsensus/trace/Span.java]: add a comment that sampling will not be re-evaluated

Comment (yurishkuro) [reply api/src/main/java/openconsensus/trace/Span.java]: more accurately, the sampling behavior depends on the implementation.

**Comment by carlosalberto** [comment]
Added a small note about sampling re-evaluation depending on the actual implementation. Don't know if this clear/enough. Let me know.

**Pull request merged**